src/i3ipc.ml

```diff
@@ -109,9 +110,28 @@ module Reply = struct
     | `String s -> Result.Ok (Unknown s)
     | j -> Result.Error ("Reply.node_layout_of_yojson: " ^ Json.to_string j)
 
+  type window_properties = {
+    class_: string option [@key "class"];
+    instance: string option;
+    title: string option;
+    transient_for: string option;
+    window_role: string option [@default None];
+  } [@@deriving of_yojson { strict = false }, show]
+
+  type id = string
```

Comment:
I would rename this type alias to `node_id` to be a bit more explicit (the name of the field in the record would stay as `id`).